src/uct/tcp/tcp_net.c

```diff
@@ -156,9 +156,9 @@ ucs_status_t uct_tcp_netif_is_default(const char *if_name, int *result_p)
     */
     while (fgets(str, sizeof(str), f) != NULL) {
         ret = sscanf(str, "%s %*x %*x %*d %*d %*d %*d %x", name, &netmask);
-        if ((ret == 3) && !strcmp(name, if_name) && (netmask == 0)) {
+        if ((ret == 2) && !strcmp(name, if_name) && (netmask == 0)) {
             *result_p = 1;
-            break;
+            return UCS_OK;
```

Comment:
you introduced a file descriptor `f` leak now, that's why you see test failures.
use `goto out;` instead of `return UCS_OK;`
and declare `out` label as follows:
```
    *result_p = 0;
out:
    fclose(f);
    return UCS_OK;
```

Reply:
Oh! I see. fixed. thanks!